File: mojo/system/embedder.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "mojo/system/embedder.h"

#include "mojo/system/core_impl.h"

namespace mojo {
namespace embedder {

void Init() {
  Core::Init(new system::CoreImpl());
}

}  // namespace embedder
}  // namespace mojo

File: chromeos/components/projector_app/resources/embedder.js
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2021 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * Gets the query string from the URL.
 * For example, if the URL is chrome://projector/annotator/abc, then query is
 * "abc".
 */
function getQuery() {
  if (!document.location.pathname) {
    return '';
  }
  const paths = document.location.pathname.split('/');
  if (paths.length < 1) {
    return '';
  }
  return paths[paths.length - 1];
}

function init() {
  // TODO(b/197343976): embed chrome-untrusted://projector/app instead.
  document.body.querySelector('iframe').src =
      'chrome-untrusted://projector/' + getQuery();
}

document.addEventListener('DOMContentLoaded', init, false);
